### PR TITLE
Add documentation how to create a new package.

### PR DESCRIPTION
**Pull request by mamokarz**

### CODE REVIEW

File: DCF_Demo.md [hunk 1]
```diff
@@ -6,8 +6,18 @@ Please make sure you have completed the Getting Started Guide for the STM32 boar
 ## Open up an instance of Azure CLI 
 You can use the Azure Portal or an instance on Powershell/bash/other local environments
 
+Device Manager current support install code from 4 different sources.
+
+0. **In memory**: Are the code that was brought to the memory using external interfaces (like JTag).
```
Comment (smcd253) [inline DCF_Demo.md]: nit: "Is the code that was brought..." #Resolved

File: DCF_Demo.md [hunk 2]
```diff
@@ -6,8 +6,18 @@ Please make sure you have completed the Getting Started Guide for the STM32 boar
...
+1. **From Blobs** (*not supported yet*): DM will download the code from Blobs using the provided URL.
+2. **From CLI** (*not supported yet*): DM will copy teh code from CLI to the memory.
```
Comment (smcd253) [inline DCF_Demo.md]: nit: "... copy the code..." #Resolved

File: DCF_Demo.md [hunk 3]
```diff
@@ -6,8 +6,18 @@ Please make sure you have completed the Getting Started Guide for the STM32 boar
...
+3. **Built in**: Are the code build with the OS.
```
Comment (smcd253) [inline DCF_Demo.md]: nit: "Is the code..." #Resolved

File: DCF_Demo.md [hunk 4]
```diff
@@ -554,4 +563,605 @@ Receive direct method: cipher.1.decrypt
 	Payload: {"src":"1P1xYWwQLAQ1ZCVMlAR8UDV0lXRAh"}
 ```
 
-</details>
\ No newline at end of file
+</details>
+
+
+### In Memory example
+The current example produces the key_vault_v1i1 package. The CMake will create the key_vault_v1i1.elf and key_vault_v1i1.bin for this package in ".\build\packages\key_vault_v1i1" directory. Before execute the commands bellow, you must shall copy the binary to the MCU internal Flash. The binary was linked to be installed in the memory 0x08050000. One of the ways to do that is using GDB commands.
```
Comment (cindydeng1998) [inline DCF_Demo.md]: nit: you **should** copy the binary to the MCU internal Flash #Resolved

**Review by cindydeng1998 — APPROVED** [review]

File: DCF_Demo.md [hunk 5]
```diff
@@ -554,4 +563,605 @@ Receive direct method: cipher.1.decrypt
...
+
+The sprinkler_v2i1 will use the same interface of sprinkler_v1i1, the main difference will be in the business logic, the version 2 will be able to control 3 different areas independently, while version 1 can control only 1 area.
+
+#### Model
+
+The model shall describe the interface, including its name and version, and a list of capabilities with the input and output arguments for each capability.
+
+Create a file called sprinkler_1_model.h with the following content in the directory "sprinkler_v2i1".
+
+```c
```
Comment (smcd253) [inline DCF_Demo.md:739]: Would it make sense to put this in a "package_to_create" directory so-as not to make the md too big?  #Pending

**Review by smcd253 — APPROVED** [review]
lgtm

**Pull request merged**